instrumentation/src/java/org/oppia/android/instrumentation/EndToEndTestNetworkModule.kt

```diff
@@ -0,0 +1,84 @@
+package org.oppia.android.instrumentation
+
+import dagger.Module
+import dagger.Provides
+import okhttp3.OkHttpClient
+import org.oppia.android.data.backends.gae.JsonPrefixNetworkInterceptor
+import org.oppia.android.data.backends.gae.NetworkApiKey
+import org.oppia.android.data.backends.gae.OppiaRetrofit
+import org.oppia.android.data.backends.gae.RemoteAuthNetworkInterceptor
+import org.oppia.android.data.backends.gae.api.ClassroomService
+import org.oppia.android.data.backends.gae.api.FeedbackReportingService
+import org.oppia.android.data.backends.gae.api.TopicService
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import javax.inject.Singleton
+
+/**
+ * Module which provides all required dependencies about network
+ *
+ * Sample resource: https://github.com/gahfy/Feed-Me/tree/unitTests
+ */
+@Module
+class EndToEndTestNetworkModule {
```

Comment:
Instead of redefining this module, could we instead introduce a NetworkConfigModule in prod that provides the endpoint URL & provide an end-to-end test variant here that provides the dev version? That would lead to less duplication with the prod configuration.

Reply:
https://github.com/oppia/oppia-android/issues/3623#issuecomment-893245336

Aren't we expecting to remove the NetworkSettings and use a single PROD_URL for the NetworkModule

Reply:
Done,
Removed the EndToEndTestNetworkModule and update the DEVELOPEMENT_URL in the NetworkSettings.kt